src/VideoCommand.cc

```diff
@@ -350,3 +360,194 @@ Json::Value FindVideo::construct_responses(
     ret[_cmd_name].swap(FindVideo);
     return ret;
 }
+
+//========= FindImage definitions =========
+
+FindFrames::FindFrames() : VideoCommand("FindFrames")
+{
+}
+
+bool FindFrames::get_interval_index (const Json::Value& cmd,
+                                     Json::ArrayIndex& op_index)
+{
+    if (cmd.isMember("operations")) {
+        const auto operations = cmd["operations"];
+        for (auto i = 0; i < operations.size(); i++) {
+            const auto op = operations[i];
+            const std::string& type = get_value<std::string>(op, "type");
+            if (type == "interval") {
+                op_index = i;
+                return true;
+            }
+        }
+    }
+    return false;
+}
+
+int FindFrames::construct_protobuf(
+    PMGDQuery& query,
+    const Json::Value& jsoncmd,
+    const std::string& blob,
+    int grp_id,
+    Json::Value& error)
+{
+    const Json::Value& cmd = jsoncmd[_cmd_name];
+
+    // We try to catch the missing attribute error before
+    // initiating a PMGD query
+    Json::ArrayIndex tmp;
+    bool is_interval = get_interval_index(cmd, tmp);
+    bool is_frames   = cmd.isMember("frames");
+
+    if (!(is_frames != is_interval)) {
+        error["status"]  = RSCommand::Error;
+        error["info"] = "Either one of 'frames' or 'operations::interval' "
+                        "must be specified";
+        return -1;
+    }
+
+    Json::Value results = get_value<Json::Value>(cmd, "results");
+    results["list"].append(VDMS_VID_PATH_PROP);
+
+    query.QueryNode(
+            get_value<int>(cmd, "_ref", -1),
+            VDMS_VID_TAG,
+            cmd["link"],
+            cmd["constraints"],
+            results,
+            get_value<bool>(cmd, "unique", false)
+            );
+
+    return 0;
+}
+
+Json::Value FindFrames::construct_responses(
+    Json::Value& responses,
+    const Json::Value& json,
+    protobufs::queryMessage &query_res,
+    const std::string &blob)
+{
+    const Json::Value& cmd = json[_cmd_name];
+
+    Json::Value ret = check_responses(responses);
+
+    auto error = [&](Json::Value& res)
+    {
+        ret[_cmd_name] = res;
+        return ret;
+    };
+
+    Json::Value& FindFrames = responses[0];
+
+    bool flag_empty = true;
+
+    for (auto& ent : FindFrames["entities"]) {
+
+        if(!ent.isMember(VDMS_VID_PATH_PROP)){
+            continue;
```

Comment:
I think this should be an assert or exception. Isn't this an internal property that must be added when a video goes into the system?